flang/runtime/misc-intrinsic.cpp

```diff
@@ -55,6 +56,36 @@ static RT_API_ATTRS void TransferImpl(Descriptor &result,
 extern "C" {
 RT_EXT_API_GROUP_BEGIN
 
+void RTDECL(Rename)(const Descriptor &path1, const Descriptor &path2,
```

Comment:
`RTDECL` -> `RTDEF`

I suggest using `#if !defined(RT_DEVICE_COMPILATION)` to guard the body of the function, as long as the device runtimes do not provide POSIX `rename`.  I would also suggest issuing an `unsupported` crash message in the else clause of the check.

Reply:
I have created https://github.com/llvm/llvm-project/pull/99445 in response to this.  Please have a look there and kindly provide a review there.